timescaledb/tutorials/grafana/visualizations/time-series.md

```diff
@@ -0,0 +1,194 @@
+
+1.  Grafana returns this graph:
+
+    <img class="main-content__illustration" src="https://assets.timescale.com/docs/images/tutorials/visualizations/time-series/simple-time-series-graph.png" alt="Screenshot of the time-series graph produced by Grafana. The graph represents the price of AMD in the past 6 hours."/>
+
+</procedure>
+
+## Create a time-series graph from pre-aggregated data
+In the previous example, we queried for all transactions of AMD stock in a 6 
+hour period which resulted in ~3 800 data points. If you query for transactions 
+of AMD stock in a 3 month period, you get around 1,500,000 data points. It's not smart 
+to graph this amount of rows every 30 seconds, which is the refresh interval 
+default. This uses a lot of CPU, memory, and network bandwidth. In extreme cases
+Grafana will freeze and require a restart.
+
+To solve this problem you can pre-aggregate your data using TimescaleDB's 
+[`time_bucket`][time_bucket] hyperfunction.
+
+<procedure>
+
+## Create a time-series graph from pre-aggregated data
+
+1.  Add the `$bucket_interval` variable of type `Interval` to the Grafana dashboard
+
+1.  In Grafana, create a new panel and add this query:
+
+    ```sql
+    SELECT time_bucket('$bucket_interval', time) AS time,
+    	AVG(price) AS price
+    FROM stocks_real_time
+    WHERE symbol = '$symbol'
+        AND time >= $__timeFrom()::timestamptz AND time < $__timeTo()::timestamptz
+    GROUP BY time_bucket('$bucket_interval', time);
+    ```   
+
+1.  With a `bucket_interval` of 30 minutes and a date range of `Last 30 days`, Grafana
+    returns this graph:
+
+    <img class="main-content__illustration" src="https://assets.timescale.com/docs/images/tutorials/visualizations/time-series/time-bucket-graph.png" alt="Screenshot of the time-series graph produced by Grafana using the `time_bucket` hyperfunction. The graph represents the price of AMD stock in the past 30 days with an interval of 30 minutes."/>
+
+    Because the stock market is only open from 9:30&nbsp;AM to 4:00&nbsp;PM on weekdays, 
+    there are large gaps in our dataset where no data is present. Grafana 
+    automatically connects the last non null to the nearest other non null value.
+    In this case, the `time_bucket` hyperfunction doesn't return a row if there 
+    is no data in the bucket. This stops you from using the `Connect null values`
+    option in the options panel.
+
+1.  In your query, replace `time_bucket` with `time_bucket_gapfill`.
+   
+    ```SQL
+    SELECT time_bucket_gapfill('$bucket_interval', time) as time,
+	    AVG(price) as price 
+    FROM stocks_real_time
+    WHERE symbol = '$symbol'
+        AND time >= $__timeFrom()::timestamptz AND time < $__timeTo()::timestamptz
+    GROUP BY time_bucket_gapfill('$bucket_interval', time);
+    ```
+
+    You can use [`time_bucket_gapfill`][time-bucket-gapfill] to circumvent the 
```

Comment:
I think this should be moved above the procedure. "but wait, there's a solution for this issue, and time_bucket_gapfill is the answer... blah, blah..."